expression/builtin_time_vec.go

```diff
@@ -183,11 +183,38 @@ func (b *builtinDayNameSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column)
 }
 
 func (b *builtinWeekDaySig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinWeekDaySig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDatetime, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err = b.args[0].VecEvalTime(b.ctx, input, buf); err != nil {
+		return err
```

Comment:
should we call `handleInvalidTimeError` here?

Reply:
This error returned by its child is not an `InvalidTimeError`, so we should return it directly here.